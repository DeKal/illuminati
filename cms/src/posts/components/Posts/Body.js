import React from 'react'
import PropTypes from 'prop-types'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import Checkbox from '@material-ui/core/Checkbox'
import { stableSort, getComparator } from 'core/utils/sort'

const Body = ({
  posts,
  page,
  rowsPerPage,
  order,
  selectedPosts,
  orderColumn,
  setSelectedPost
}) => (
  <TableBody>
    {stableSort(posts, getComparator(order, orderColumn))
      .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
      .map((row, index) => {
        const isItemSelected = selectedPosts.indexOf(row.name) !== -1
        const labelId = `enhanced-table-checkbox-${index}`

        return (
          <TableRow
            hover
            onClick={() => setSelectedPost(row.name)}
            role="checkbox"
            aria-checked={isItemSelected}
            tabIndex={-1}
            key={row.name}
            selected={isItemSelected}
          >
            <TableCell padding="checkbox">
              <Checkbox
                checked={isItemSelected}
                inputProps={{ 'aria-labelledby': labelId }}
              />
            </TableCell>
            <TableCell component="th" id={labelId} scope="row" padding="none">
              {row.name}
            </TableCell>
            <TableCell align="right">{row.calories}</TableCell>
            <TableCell align="right">{row.fat}</TableCell>
            <TableCell align="right">{row.carbs}</TableCell>
            <TableCell align="right">{row.protein}</TableCell>
          </TableRow>
        )
      })}
    {renderEmptyRows(page, rowsPerPage, posts.length)}
  </TableBody>
)

export default Body

Body.propTypes = {
  posts: PropTypes.array,
  page: PropTypes.number,
  rowsPerPage: PropTypes.number,
  order: PropTypes.string,
  orderColumn: PropTypes.string,
  selectedPosts: PropTypes.array,
  setSelectedPost: PropTypes.func
}

const renderEmptyRows = (page, rowsPerPage, postLength) => {
  const emptyRows =
    rowsPerPage - Math.min(rowsPerPage, postLength - page * rowsPerPage)
  return emptyRows ? (
    <TableRow style={{ height: 53 * emptyRows }}>
      <TableCell colSpan={6} />
    </TableRow>
  ) : null
}
