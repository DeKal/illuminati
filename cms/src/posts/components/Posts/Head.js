import React from 'react'
import PropTypes from 'prop-types'
import { styled } from '@material-ui/core/styles'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableSortLabel from '@material-ui/core/TableSortLabel'
import Checkbox from '@material-ui/core/Checkbox'

const EnhancedTableHead = ({
  order,
  orderColumn,
  numSelected,
  rowCount,
  setOrder,
  setOrderColumn,
  selectAllPosts,
  clearSelected
}) => {
  const createSortHandler = property => () => {
    const isAsc = orderColumn === property && order === 'asc'
    setOrder(isAsc ? 'desc' : 'asc')
    setOrderColumn(property)
  }

  return (
    <TableHead>
      <TableRow>
        <TableCell padding="checkbox">
          <Checkbox
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={rowCount > 0 && numSelected === rowCount}
            onChange={event => {
              if (event.target.checked) {
                selectAllPosts()
                return
              }
              clearSelected()
            }}
            inputProps={{ 'aria-label': 'select all desserts' }}
          />
        </TableCell>
        {headCells.map(headCell => (
          <TableCell
            key={headCell.id}
            align={headCell.numeric ? 'right' : 'left'}
            padding={headCell.disablePadding ? 'none' : 'default'}
            sortDirection={orderColumn === headCell.id ? order : false}
          >
            <TableSortLabel
              active={orderColumn === headCell.id}
              direction={orderColumn === headCell.id ? order : 'asc'}
              onClick={createSortHandler(headCell.id)}
            >
              {headCell.label}
              {orderColumn === headCell.id ? (
                <VisuallyHiddenSpan>
                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                </VisuallyHiddenSpan>
              ) : null}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  )
}

EnhancedTableHead.propTypes = {
  numSelected: PropTypes.number,
  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
  orderColumn: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
  setOrder: PropTypes.func,
  setOrderColumn: PropTypes.func,
  selectAllPosts: PropTypes.func,
  clearSelected: PropTypes.func
}

export default EnhancedTableHead

const headCells = [
  {
    id: 'title',
    numeric: false,
    disablePadding: true,
    label: 'Title'
  },
  { id: 'author', numeric: false, disablePadding: false, label: 'Author' },
  { id: 'tag', numeric: false, disablePadding: false, label: 'Tags' },
  { id: 'commentNum', numeric: true, disablePadding: false, label: 'Comments' },
  { id: 'date', numeric: false, disablePadding: false, label: 'Date' }
]

const VisuallyHiddenSpan = styled('span')({
  border: 0,
  clip: 'rect(0 0 0 0)',
  height: 1,
  margin: -1,
  overflow: 'hidden',
  padding: 0,
  position: 'absolute',
  top: 20,
  width: 1
})
