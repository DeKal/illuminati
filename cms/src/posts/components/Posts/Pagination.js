import React from 'react'
import PropTypes from 'prop-types'
import TablePagination from '@material-ui/core/TablePagination'
import { calcMaxRowsPerPage } from 'core/utils/misc'

const Pagination = ({
  postCount,
  rowsPerPage,
  page,
  changePage,
  changeRowsPerPage
}) => (
  <TablePagination
    rowsPerPageOptions={calcMaxRowsPerPage(postCount)}
    component="div"
    count={postCount}
    rowsPerPage={rowsPerPage}
    page={page}
    onChangePage={(_, newPage) => changePage(newPage)}
    onChangeRowsPerPage={event =>
      changeRowsPerPage(parseInt(event.target.value, 10))
    }
  />
)

Pagination.propTypes = {
  postCount: PropTypes.number,
  rowsPerPage: PropTypes.number,
  page: PropTypes.number,
  changePage: PropTypes.func,
  changeRowsPerPage: PropTypes.func
}

export default Pagination
