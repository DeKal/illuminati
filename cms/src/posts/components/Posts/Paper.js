import { withStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'

const StylePaper = withStyles(theme => ({
  root: {
    width: '100%',
    marginBottom: theme.spacing(2)
  }
}))(Paper)

export default StylePaper
