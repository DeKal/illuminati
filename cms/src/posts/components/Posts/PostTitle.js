import { withStyles } from '@material-ui/core/styles'
import TableCell from '@material-ui/core/TableCell'

const StyleTableCell = withStyles(theme => ({
  root: {
    width: '50%',
    cursor: 'pointer',
    '&:hover': {
      color: theme.colors.primary
    }
  }
}))(TableCell)

export default StyleTableCell
