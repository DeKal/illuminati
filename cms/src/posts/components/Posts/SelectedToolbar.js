import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import Tooltip from '@material-ui/core/Tooltip'
import DeleteIcon from '@material-ui/icons/Delete'

const SelectedToolbar = ({ classes, numSelected }) => (
  <Toolbar
    className={clsx(classes.root, {
      [classes.highlight]: numSelected
    })}
  >
    {numSelected ? (
      <Typography className={classes.title} color="inherit" variant="subtitle1">
        {numSelected} selected
      </Typography>
    ) : (
      <Typography className={classes.title} variant="h6">
        All Posts
      </Typography>
    )}

    {numSelected ? (
      <Tooltip title="Delete">
        <IconButton aria-label="delete">
          <DeleteIcon />
        </IconButton>
      </Tooltip>
    ) : null}
  </Toolbar>
)

SelectedToolbar.propTypes = {
  numSelected: PropTypes.number.isRequired,
  classes: PropTypes.shape({
    root: PropTypes.object,
    title: PropTypes.object,
    highlight: PropTypes.object
  })
}

export default SelectedToolbar
