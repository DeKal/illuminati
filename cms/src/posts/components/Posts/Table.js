import { withStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'

const CustomTable = withStyles(theme => ({
  root: {
    minWidth: theme.postsTableWidth
  }
}))(Table)

export default CustomTable
