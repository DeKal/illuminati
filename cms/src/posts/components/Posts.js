import React from 'react'
import PropTypes from 'prop-types'
import TableContainer from '@material-ui/core/TableContainer'
import Container from 'core/components/Container'
import Head from 'posts/containers/Head'
import Pagination from 'posts/containers/Pagination'
import Body from 'posts/containers/Body'
import SelectedToolbar from 'posts/containers/SelectedToolbar'
import Paper from 'posts/components/Posts/Paper'
import Table from 'posts/components/Posts/Table'

const Posts = ({ posts, selectedPosts }) => {
  const numSelected = selectedPosts.length
  const rowCount = posts.length
  return (
    <Container>
      <Paper>
        <SelectedToolbar numSelected={numSelected} />
        <TableContainer>
          <Table
            aria-labelledby="tableTitle"
            size={'medium'}
            aria-label="enhanced table"
            style={{ marginBottom: '15px', overflow: 'auto' }}
          >
            <Head numSelected={numSelected} rowCount={rowCount} />
            <Body posts={posts} selectedPosts={selectedPosts} />
          </Table>
        </TableContainer>
        <Pagination postCount={rowCount} />
      </Paper>
    </Container>
  )
}

export default Posts

Posts.propTypes = {
  posts: PropTypes.array,
  selectedPosts: PropTypes.array,
  fetchPostSummary: PropTypes.func
}
