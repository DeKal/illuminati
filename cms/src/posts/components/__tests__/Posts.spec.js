import * as React from 'react'
import Posts from '../Posts'
import { shallow } from 'enzyme'

const posts = [
  {
    slug: 'getting-started-with-java',
    title: 'Getting Started With Java',
    time: '06/01/2020',
    tag: ['Web Development', 'Java'],
    author: 'Phat Ho',
    commentNum: 0
  },
  {
    slug: 'get-startted-with-spring',
    title: 'Get Startted With Spring MVC',
    time: '06/01/2020',
    tag: ['Java Spring', 'Course'],
    author: 'Phat Ho',
    commentNum: 0
  }
]

describe('Posts', () => {
  it('should render Posts with no error', function() {
    const wrap = shallow(<Posts posts={posts} selectedPosts={[]} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render Loading', function() {
    const wrap = shallow(<Posts posts={[]} selectedPosts={[]} />)
    expect(wrap).toMatchSnapshot()
  })
})
