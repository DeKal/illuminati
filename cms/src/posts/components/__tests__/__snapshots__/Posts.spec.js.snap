// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts should render Loading 1`] = `<Loading />`;

exports[`Posts should render Posts with no error 1`] = `
<Styled(div)>
  <WithStyles(WithStyles(ForwardRef(Paper)))>
    <SelectedToolbarContainer
      numSelected={0}
    />
    <WithStyles(ForwardRef(TableContainer))>
      <WithStyles(WithStyles(ForwardRef(Table)))
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        size="medium"
        style={
          Object {
            "marginBottom": "15px",
            "overflow": "auto",
          }
        }
      >
        <Connect(EnhancedTableHead)
          numSelected={0}
          rowCount={2}
        />
        <Connect(withRouter(Body))
          posts={
            Array [
              Object {
                "author": "Phat Ho",
                "commentNum": 0,
                "slug": "getting-started-with-java",
                "tag": Array [
                  "Web Development",
                  "Java",
                ],
                "time": "06/01/2020",
                "title": "Getting Started With Java",
              },
              Object {
                "author": "Phat Ho",
                "commentNum": 0,
                "slug": "get-startted-with-spring",
                "tag": Array [
                  "Java Spring",
                  "Course",
                ],
                "time": "06/01/2020",
                "title": "Get Startted With Spring MVC",
              },
            ]
          }
          selectedPosts={Array []}
        />
      </WithStyles(WithStyles(ForwardRef(Table)))>
    </WithStyles(ForwardRef(TableContainer))>
    <Connect(Pagination)
      postCount={2}
    />
  </WithStyles(WithStyles(ForwardRef(Paper)))>
</Styled(div)>
`;
